apis/definition.yaml

```diff
@@ -0,0 +1,74 @@
+apiVersion: apiextensions.crossplane.io/v1
+kind: CompositeResourceDefinition
+metadata:
+  name: xflux.gitops.platform.upbound.io
+spec:
+  group: gitops.platform.upbound.io
+  names:
+    kind: XFlux
+    plural: xflux
+  versions:
+    - name: v1alpha1
+      served: true
+      referenceable: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                parameters:
+                  type: object
+                  description: Flux configuration parameters.
+                  properties:
+                    id:
+                      type: string
+                      description: ID of this Cluster that other objects will use to refer to it
+                    deletionPolicy:
+                      description: Delete the external resources when the Claim/XR is deleted. Defaults to Delete
+                      enum:
+                        - Delete
+                        - Orphan
+                      type: string
+                      default: Delete
+                    kubeConfigSecretRef:
+                      type: object
+                      properties:
+                        name:
+                          type: string
+                        key:
+                          type: string
+                          default: kubeconfig
+                    operators:
+                      type: object
+                      properties:
+                        flux:
+                          type: object
+                          properties:
+                            version:
+                              type: string
+                        flux-sync:
+                          type: object
+                          properties:
+                            version:
+                              type: string
+                    source:
+                      type: object
+                      properties:
+                        git:
+                          type: object
+                          properties:
+                            interval:
+                              type: string
```

Comment:
might be good to extend this gitops API fields with `description` for self-documenting purposes

Reply:
add description for all fields